helix-term/src/ui/overlay.rs

```diff
@@ -0,0 +1,117 @@
+use crossterm::event::Event;
+use helix_core::Position;
+use helix_view::{
+    graphics::{CursorKind, Rect},
+    Editor,
+};
+use tui::buffer::Buffer;
+
+use crate::compositor::{Component, Context, EventResult};
+
+/// Contains a component placed in the center of the parent component
+#[derive(Debug)]
+pub struct Overlay<T> {
+    /// Child component
+    pub content: T,
+    /// Fixed margin around the component
+    pub margin: Margin4,
+    /// Value between 0 and 1 that indicates how much of the vertical space is used
+    pub vertical_size: f32,
+    /// Value between 0 and 1 that indicates how much of the horizontal space is used
+    pub horizontal_size: f32,
+}
+
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Margin4 {
```

Comment:
There's a margin type that can be used with `rect.inner(margin)` https://github.com/helix-editor/helix/blob/e2d2f19fd02419b8d144ae7a8a359d31bd773af7/helix-view/src/graphics.rs#L28-L32

Reply:
The purpose of this type is to add a 2-row margin at the bottom (because there's the status bar) but not at the top, so the picker appears vertically centered:

![helix-fp](https://user-images.githubusercontent.com/15658558/152381907-3693b8e7-c524-4333-a12b-415ab7701f78.png)

Is this not desired?

Reply:
It's a workaround, but I'd just shrink the `area` that's passed in by `2`. I think we have a method for that (`clip_bottom`)

Reply:
I removed the `Margin4` struct.